genesis-builder/src/lib.rs

```diff
@@ -658,25 +656,34 @@ impl GenesisBuilder {
         debug!("Accounts:");
         debug!(?accounts);
 
-        let block_path = directory.as_ref().join("block.dat");
+        let block_path = directory.join("block.dat");
         info!(path = %block_path.display(), "Writing block to");
-        let mut file = OpenOptions::new()
-            .create(true)
-            .truncate(true)
-            .write(true)
-            .open(&block_path)?;
-        block.serialize_to_writer(&mut file)?;
+        fs::write(block_path, block.serialize_to_vec())?;
+
+        let decompressed_path = directory.join("decompressed_keys.dat");
+        if let Block::Macro(MacroBlock {
+            header:
+                MacroHeader {
+                    validators: Some(validators),
+                    ..
+                },
+            ..
+        }) = &block
+        {
+            let mut decompressed = Vec::new();
+            for key in validators.voting_keys() {
+                decompressed.extend_from_slice(&key.trusted_serialize());
+            }
+            fs::write(decompressed_path, decompressed)?;
+        } else {
+            unreachable!("genesis block must be an election block");
+        }
```

Comment:
You may be able to use `block.validators().expect()` here as that will give you the `Validators` as well and is a bit shorter and potentially better readable.

Reply:
Done.